test/parallel/test-worker-fshandles-closed.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const {
+  setImmediate,
+} = require('timers/promises');
+const { isMainThread, parentPort, Worker } = require('worker_threads');
+
+const MAX_ITERATIONS = 20;
+const MAX_THREADS = 10;
+
+if (isMainThread) {
+  function spinWorker(iter) {
+    const w = new Worker(__filename, { workerData: iter });
+    w.on('message', common.mustCall((msg) => {
+      assert.strictEqual(msg, 'terminate');
+      w.terminate();
+    }));
+
+    w.on('exit', common.mustCall(() => {
+      if (iter < MAX_ITERATIONS)
+        spinWorker(++iter);
+    }));
+  }
+
+  for (let i = 0; i < MAX_THREADS; i++) {
+    spinWorker(0);
+  }
+} else {
+  // No need to require it from the main thread as it's not used there.
```

Comment:
Same is true for `timers/promises`, yet it's not there. I'm not convinced it's a good enough reason to split the require calls, please move it to the top level.

Reply:
> Same is true for timers/promises, yet it's not there.

You're right, I missed moving that to the worker thread scope.

> I'm not convinced it's a good enough reason to split the require calls, please move it to the top level.

I'd rather not load unnecessary modules in threads that don't use them, but well, it's just a test. I'll do as you prefer.